crates/bevy_winit/src/state.rs

```diff
@@ -912,3 +912,112 @@ pub(crate) fn react_to_scale_factor_change(
         });
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use bevy_app::Update;
+
+    use super::*;
+
+    #[test]
+    fn test_react_to_scale_factor_change_with_changed_scale_factor() {
+        let (mut app, window_entity) = setup_react_to_scale_factor_change_test_app(1.0, 2.0);
+        app.update();
+
+        let window = app.world().get::<Window>(window_entity);
+        assert_eq!(window.unwrap().resolution.scale_factor(), 2.0);
+
+        let window_backend_scale_factor_changed = app
+            .world()
+            .resource::<Events<WindowBackendScaleFactorChanged>>();
```

Comment:
You could simplify things here by just directly using `iter_current_update_events` or `drain` maybe? Same thing with the other test.